### PR TITLE
Add neverthrow error handling + update docs + add example

### PR DESCRIPTION
**Pull request by TheLDB**
- Switched all error handling to neverthrow, which provides Rust-style Result<T, E> in Typescript for type-safe error handling
- Added a large example file that tests all the modules w/ a quick start
- Updated docs w/ some basic getting started stuff

### CODE REVIEW

**Review by neverabsolute — APPROVED** [review]

**Comment by neverabsolute** [comment]
https://github.com/jacc/whoopkit/assets/58273860/5b401c9c-218e-4229-8a65-32c780e34ce9

check out this epearl clutch btw

**Comment by neverabsolute** [comment]
https://github.com/jacc/whoopkit/assets/58273860/8085c6c1-3fdc-468e-b98d-aa88d73cc7e1

check out this clip, sent that car flying!

File: helpers/neverthrowFetch.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { ResultAsync } from "neverthrow";
+import { WhoopAsyncResult } from "../types/neverthrow";
+import { requestHeaders } from "../types/constants";
+
+export type neverthrowFetchProps = {
+  init?: RequestInit;
+  error?: string;
+};
+
+export const neverthrowFetch = async <T>(
```
Comment (jacc) [inline helpers/neverthrowFetch.ts]: can we call this safeFetch or something else other than neverthrow?

File: modules/health.ts [hunk 2]
```diff
@@ -6,157 +6,139 @@ TODO:
...
 
 export class WhoopKitHealth {
   constructor(private accessToken: string) {
     this.accessToken = accessToken;
   }
 
   // TODO: type this
-  async getStress(date?: string): Promise<any> {
-    const request = await fetch(
+  async getStress(date?: string): WhoopAsyncResult<any> {
```
Comment (jacc) [inline modules/health.ts]: this type is an easy one, just all strings

File: modules/health.ts [hunk 3]
```diff
@@ -6,157 +6,139 @@ TODO:
...
+      score: res.gauge.gauge_score_display,
+      text: res.gauge.gauge_subtext_display,
+      state: res.stress_state,
+      last_updated: res.last_updated_display,
+    }));
   }
 
-  async getAverageHRV(date?: string): Promise<any> {
-    const request = await fetch(
+  async getAverageHRV(date?: string): WhoopAsyncResult<any> {
```
Comment (jacc) [inline modules/health.ts]: all getAverage functions return the same body, so the type can be reused

**Review by jacc — CHANGES_REQUESTED** [review]

File: helpers/neverthrowFetch.ts (hunk 1 above)
Comment (TheLDB) [reply helpers/neverthrowFetch.ts]: nuh uh

Comment (TheLDB) [reply helpers/neverthrowFetch.ts]: ok fixed, renamed to safeFetch

File: modules/health.ts (hunk 3 above)
Comment (TheLDB) [reply modules/health.ts]: made a generic `AverageResponse` type instead

File: modules/health.ts (hunk 2 above)
Comment (TheLDB) [reply modules/health.ts]: resolved

**Pull request merged**